### PR TITLE
Add component references system

### PR DESCRIPTION
**Pull request by jrising**
Here is the system for component references-- so that we can write, e.g.,
connectparameter(model, :MyComponent, :myparameter, :YourComponent, :yourparameter)
as
mycomponent[:myparameter] = yourcomponent[:yourparameter]

### CODE REVIEW

**Comment by jrising** [comment]
Have you seen this before?  Only 32 bit windows failing, for a very simple model?  I thought it had to do with my use of 'pi' as a test number, but no.

File: test/test_references.jl [hunk 1]
```diff
@@ -0,0 +1,33 @@
+using Base.Test
+using Mimi
+
+@defcomp Foo begin
+	input = Parameter()
+	intermed = Variable(index=[time])
+end
+
+function timestep(c::Foo, tt::Int)
```
Comment (davidanthoff) [inline test/test_references.jl]: Try to remove the type annotation on ``tt`` here, i.e. change this line to ``function timestep(c::Foo, tt)``. I think what is happening is that ``Int`` defaults to ``Int64`` on x64 platforms, but to ``Int32`` on x86 systems. But Mimi calls ``timestep`` always with a ``Int64`` for the second parameter. So when you declare with ``Int``, multi dispatch will fall back to the generic timestep on x86 systems, and your timestep function is never called.

File: test/test_references.jl [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    c.Variables.intermed[tt] = c.Parameters.input
+end
+
+@defcomp Bar begin
+	intermed = Parameter(index=[time])
+	output = Variable(index=[time])
+end
+
+function timestep(c::Bar, tt::Int)
```
Comment (davidanthoff) [inline test/test_references.jl]: Same here.

**Comment by davidanthoff** [comment]
Great, this is really cool!

**Pull request merged**